core/templates/dev/head/domain/topic/TopicRightsObjectFactory.js

```diff
@@ -75,13 +75,17 @@ oppia.factory('TopicRightsObjectFactory', [
     TopicRights.prototype.copyFromTopicRights = function(otherTopicRights) {
       this._published = otherTopicRights.isPublished();
       this._canEditTopic = otherTopicRights.canEditTopic();
-      this._canPublishTopic = otherTopicRights.getCanPublishTopic();
+      this._canPublishTopic = otherTopicRights.canPublishTopic();
     };
 
     // Create a new, empty topic rights object. This is not guaranteed to
     // pass validation tests.
     TopicRights.createEmptyTopicRights = function() {
-      return new TopicRights({});
+      return new TopicRights({
```

Comment:
Just to check: why an object with fields that have a null value, rather than saying `topicRights = null`? What are the pros and cons?

Also, for a topic rights object like this, what happens if you call topicRights.markTopicAsUnpublished(), topicRights.canEdit(), etc.? That case needs to be handled explicitly, but then it's going to introduce a lot of clutter. So consider the alternative of having topicRights just not exist at all at the outset; what happens then?

If you do need a topicRights early on then maybe createDefault does need to be a valid rights object but with the most restrictive settings possible.

Anyway -- this is something to figure out now, since you'll need to reuse this pattern in the future.

Reply:
Yep, that sounds good to me. Having the actual information from the get-go would be better.

Reply:
Done, added a createInterstitialObject function that returns a valid but highly restrictive object for rights and one with placeholder values for topic object.